Caching: Add `CACHE_VERSION` attribute to `CalcJob` and `Parser` (#6328)
aiidateam/aiida-core: docs/source/topics/provenance/caching.rst

        }
    ]

.. versionchanged:: 2.6
    Version information removed from hash computation

    Up until v2.6, the objects used to compute the hash of a ``ProcessNode`` included the ``version`` attribute.

>     This attribute stored a dictonary of the version of the installed version of ``aiida-core`` at the time of creation, as well as the plugin package, if relevant.

Reviewer: The version attribute still exist so this should use past tense
```suggestion
    This attribute stores a dictionary of the installed versions of the ``aiida-core`` and plugin packages (if relevant)  at the time of creation.
```

>     When the caching mechanism was first introduced, this information was added intentionally added to the hash to err on the safe side and prevent false positives as much as possible.

Reviewer: ```suggestion
    When the caching mechanism was first introduced, this information was added to the hash intentionally to err on the safe side and prevent false positives as much as possible.
```

The hashing of *Process nodes* is fixed and can only be influenced indirectly via the hashes of their inputs.
For implementation details of the hashing mechanism for process nodes, see :ref:`here <internal_architecture:engine:caching>`.

.. _topics:provenance:caching:control-hashing:calcjobs-parsers:

Calculation jobs and parsers
............................

.. versionadded:: 2.6
    Resetting the calculation job cache

>     When the implementation of a ``CalcJob`` or ``Parser`` plugin changes significantly, it can be the case that for identical inputs, significantly different outputs are expected.

Reviewer: Might be nice to add a few practical examples here to make sure the plugin developers understand when they should bump the cache version: e.g.

- Adding an output node to the parser
- Changing an existing output node (given identical input nodes, identical log files).
- Changing defaults in Calcjob inputs for a QM computation that are part of the QM input file, but not part of AiiDA input nodes (although I guess this should be discouraged in general?). I could imagine e.g. tightening WF convergence thresholds or something like that?
